File: app/src/main/java/com/example/cbanda/achieve/Models/db/GoalDao.java
package com.example.cbanda.achieve.Models.db;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

/**
 * Created by CBanda on 2018/02/07.
 */
@Dao
public interface GoalDao {

    @Query("SELECT * from goals")
    LiveData<List<Goal>> getListGoals();

    @Insert(onConflict = REPLACE)
    void addGoal(Goal goal);

}

File: app/src/main/java/com/example/cbanda/achieve/models/db/GoalDao.kt
package com.example.cbanda.achieve.models.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query


@Dao
interface GoalDao {

    @get:Query("SELECT * from goals")
    val listGoals: LiveData<List<Goal>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addGoal(goal: Goal)

    @Query("SELECT * FROM goals WHERE id =:id")
    fun getGoalById(id: Int): LiveData<Goal>
}
